src/fheroes2/editor/editor_interface.cpp

```diff
@@ -1062,6 +1063,12 @@ namespace Interface
 
             setObjectOnTileAsAction( tile, groupType, _editorPanel.getSelectedObjectType() );
 
+            Heroes * hero = world.GetHeroForHire( Race::IndexToRace( static_cast<int>( objectInfo.metadata[1] ) ) );
+            if ( hero ) {
+                hero->SetCenter( tilePos );
+                hero->SetColor( static_cast<int>( color ) );
```

Comment:
We should convert color index here:
```suggestion
                hero->SetColor( static_cast<int>( Color::IndexToColor( static_cast<int>( color ) ) ) );
```